plugins/inputs/mesos/mesos_test.go

```diff
@@ -290,7 +338,144 @@ func TestMesosMaster(t *testing.T) {
 		t.Errorf(err.Error())
 	}
 
-	acc.AssertContainsFields(t, "mesos", masterMetrics)
+	expectedUntaggedMetrics := map[string]interface{}{}
+	for k, v := range masterMetrics {
+		parts := strings.Split(k, "/")
+		if !strings.HasPrefix(k, "master/frameworks/") && (!strings.HasPrefix(k, "allocator/") || len(parts) <= 5) {
+			expectedUntaggedMetrics[k] = v
+		}
+	}
+
+	acc.AssertContainsFields(t, "mesos", expectedUntaggedMetrics)
+
+	frameworkFields := []map[string]interface{}{
+		// framework offers
+		{
+			"master/frameworks/calls_total":      masterMetrics["master/frameworks/marathon/abc-123/calls"],
+			"master/frameworks/events_total":     masterMetrics["master/frameworks/marathon/abc-123/events"],
+			"master/frameworks/operations_total": masterMetrics["master/frameworks/marathon/abc-123/operations"],
+			"master/frameworks/subscribed_total": masterMetrics["master/frameworks/marathon/abc-123/subscribed"],
+			"master/frameworks/offers/sent":      masterMetrics["master/frameworks/marathon/abc-123/offers/sent"],
+		},
+		{
+			"master/frameworks/tasks/active": masterMetrics["master/frameworks/marathon/abc-123/tasks/active/task_killing"],
+		},
+		{
+			"master/frameworks/tasks/active":   masterMetrics["master/frameworks/marathon/abc-123/tasks/active/task_dropped"],
+			"master/frameworks/tasks/terminal": masterMetrics["master/frameworks/marathon/abc-123/tasks/terminal/task_dropped"],
+		},
+		{
+			"master/frameworks/roles/suppressed": masterMetrics["master/frameworks/marathon/abc-123/roles/*/suppressed"],
+		},
+		{
+			"master/frameworks/calls": masterMetrics["master/frameworks/marathon/abc-123/calls/accept"],
+		},
+		{
+			"master/frameworks/events": masterMetrics["master/frameworks/marathon/abc-123/events/error"],
+		},
+		{
+			"master/frameworks/operations": masterMetrics["master/frameworks/marathon/abc-123/operations/create"],
+		},
+		// allocator
+		{
+			"allocator/mesos/roles/shares/dominant":      masterMetrics["allocator/mesos/roles/*/shares/dominant"],
+			"allocator/mesos/offer_filters/roles/active": masterMetrics["allocator/mesos/offer_filters/roles/*/active"],
+		},
+		{
+			"allocator/mesos/quota/roles/resources/offered_or_allocated": masterMetrics["allocator/mesos/quota/roles/*/resources/disk/offered_or_allocated"],
+			"allocator/mesos/quota/roles/resources/guarantee":            masterMetrics["allocator/mesos/quota/roles/*/resources/disk/guarantee"],
+		},
+		{
+			"allocator/mesos/quota/roles/resources/guarantee": masterMetrics["allocator/mesos/quota/roles/*/resources/mem/guarantee"],
+		},
+	}
+
+	frameworkTags := []map[string]string{
+		// framework offers
+		{
+			"server":         m.masterURLs[0].Hostname(),
+			"url":            masterTestServer.URL,
+			"role":           "master",
+			"state":          "leader",
+			"framework_name": "marathon",
```

Comment:
what if tag contains whitespace?

Reply:
hmm I think we're expecting the metrics input (mesos metrics in this case) to contain correct metric names, so I'm not sure if we should handle such errors i.e. I don't expect to receive something like `master/frameworks//<id>/subscribed`. but I'm not certain - if that's not a reasonable expectation I can exclude empty tags from being sent on the metric

Reply:
@philipnrmn What do you think? I think we should be prepared for incorrect data.

Reply:
This is a good point! However the tag is url-encoded by Mesos, so I think we're safe